target/linux/generic/files/drivers/net/phy/rtl8366_smi.c

```diff
@@ -1558,6 +1558,12 @@ int rtl8366_smi_probe_of(struct platform_device *pdev, struct rtl8366_smi *smi)
 		goto try_gpio;
 	}
 
+	of_property_read_u32(np, "phy_id", &smi->phy_id);
```

Comment:
Here's an unrelated issue... of_property_read_u32 returns a negative value on failure. The "if" in the line following must therefore check for the return value and not smi->phy_id...

Reply:
Done